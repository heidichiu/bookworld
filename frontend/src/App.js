const App = () => {
  return <div>First App component</div>;
};

export default App;
