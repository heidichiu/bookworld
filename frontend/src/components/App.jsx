import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import BookContainer from "./book/BookContainer";
import Layout from "./layout/Layout";
import Login from "./user/Login";
import { SnackbarProvider } from "notistack";
import Auth from "./Auth";
import Register from "./user/Register";
import SingleBook from "./book/SingleBook";
import BookReviewForm from "./book/BookReviewForm";

const App = () => {
  return (
    <SnackbarProvider maxSnack={3}>
      <Router>
        <Layout>
          <Switch>
            <Route path="/login">
              <Login />
            </Route>
            <Route path="/register">
              <Register />
            </Route>
            <Route exact path="/books/:bookId">
              <SingleBook />
            </Route>
            <Route exact path="/reviews">
              <BookReviewForm />
            </Route>
            <Route exact path="/">
              <Auth>
                <BookContainer />
              </Auth>
            </Route>
          </Switch>
        </Layout>
      </Router>
    </SnackbarProvider>
  );
};

export default App;
