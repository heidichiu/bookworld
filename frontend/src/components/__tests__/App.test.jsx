import { render } from "@testing-library/react";
import App from "../App";
import renderWithRedux from "../../util/testUtil";

describe("App component", () => {
  it("should render app without error", () => {
    const { asFragment } = renderWithRedux(<App />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
