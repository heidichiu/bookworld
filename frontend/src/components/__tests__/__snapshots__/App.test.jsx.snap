// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app without error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Book World
          </h6>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-bookContainer-4"
      >
        <div
          class="MuiBox-root MuiBox-root-11 makeStyles-bookFilter-5"
        >
          <div
            class="MuiPaper-root makeStyles-bookFilterPaper-6 MuiPaper-elevation1 MuiPaper-rounded"
          >
            Book Filter
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-12 makeStyles-bookList-7"
        >
          <div
            class="MuiBox-root MuiBox-root-13"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              data-testid="book-loader"
              style="width: 80%; height: 200px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
